packages/web-components/@sfx/product/test/unit/product.test.ts

```diff
@@ -0,0 +1,231 @@
+import { TemplateResult, html } from 'lit-element';
+import { Base } from '@sfx/base';
+
+import { expect, stub, spy } from '../utils';
+
+import Product from '../../src/product';
+import Variant from '../../src/variant';
+
+describe('Product Component', () => {
+  let component;
+
+  beforeEach(() => {
+    component = new Product();
+  });
+
+  describe('constructor', () => {
+    it('should extend the Base class', () => {
+      expect(component).to.be.an.instanceof(Base);
+    });
+
+    describe('product property', () => {
+      it('should have a default value of "full"', () => {
+        expect(component.product).to.exist;
+      });
+    });
+  });
+
+  describe('connectedCallback', () => {
+    it('should call its super connectedCallback', () => {
+      const connectedCallback = stub(Base.prototype, 'connectedCallback');
+
+      component.connectedCallback();
+      expect(connectedCallback).to.have.been.called;
+    });
+
+    it('should add an eventListener to the window', () => {
+      const addEventListener = spy(window, 'addEventListener');
+
+      component.connectedCallback();
+      expect(addEventListener).to.have.been.calledWith(
+        'sfx::change_product_variant',
+        component.updateVariant
+      );
+    });
+  });
+
+  describe('disconnectedCallback', () => {
+    it('should call its super disconnectedCallback', () => {
+      const disconnectedCallback = stub(Base.prototype, 'disconnectedCallback');
+
+      component.disconnectedCallback();
+      expect(disconnectedCallback).to.have.been.called;
+    });
+
+    it('should remove eventListener from the window', () => {
+      const removeEventListener = spy(window, 'removeEventListener');
+
+      component.disconnectedCallback();
+      expect(removeEventListener).to.have.been.calledWith(
+        'sfx::change_product_variant',
+        component.updateVariant
+      );
+    });
+  });
+
+  describe('updateVariant', () => {
+    it('should update the product property with new info', () => {
+      const event = {
+        detail: { name: 'Foo', price: 12.34 }
+      };
+      component.updateVariant(event);
+      expect(component.product.name).to.equal('Foo');
+      expect(component.product.price).to.equal(12.34);
+    });
+  });
+
+  describe('additionalInfo', () => {
+    it('should return an empty array if there are no additional properties', () => {
+      const result = component.additionalInfo();
+      expect(result).to.eql([]);
+    });
+
+    it('should return an array of TemplateResults there are additional properties', () => {
+      component.product = {
+        name: '',
+        salePrice: 0,
+        promo: '30% off'
+      };
+
+      const result = component.additionalInfo();
+
+      expect(result[1]).to.be.an.instanceOf(TemplateResult);
```

Comment:
I think we should assert the entire array. You can make multiple assertions on the length and the individual items, or make use of [Sinon matchers](https://sinonjs.org/releases/latest/matchers/).